### PR TITLE
layers: Cleanup PreCallValidateCreateImageView

### PR DESCRIPTION
**Pull request by juan-lunarg**
Return early to avoids nesting majority of logic

Minor C++17 cleanup

Minor const correctness

Use image_usage consistently, since it can be modified by VkImageViewUsageCreateInfo in the pNext chain

### CODE REVIEW

File: layers/core_checks/image_validation.cpp [hunk 1]
```diff
@@ -1581,533 +1581,524 @@ bool CoreChecks::ValidateImageViewFormatFeatures(const IMAGE_STATE *image_state,
 }
 
 bool CoreChecks::PreCallValidateCreateImageView(VkDevice device, const VkImageViewCreateInfo *pCreateInfo,
-                                                const VkAllocationCallbacks *pAllocator, VkImageView *pView) const {
+                                                [[maybe_unused]] const VkAllocationCallbacks *pAllocator,
+                                                [[maybe_unused]] VkImageView *pView) const {
```
Comment (ncesario-lunarg) [inline layers/core_checks/image_validation.cpp:1585]: Would it be preferable to just remove the variable name declarations?

File: layers/core_checks/image_validation.cpp [hunk 2]
```diff
@@ -1581,533 +1581,524 @@ bool CoreChecks::ValidateImageViewFormatFeatures(const IMAGE_STATE *image_state,
...
+                                                VK_IMAGE_USAGE_VIDEO_DECODE_DST_BIT_KHR | VK_IMAGE_USAGE_VIDEO_DECODE_DPB_BIT_KHR |
+                                                VK_IMAGE_USAGE_VIDEO_ENCODE_SRC_BIT_KHR | VK_IMAGE_USAGE_VIDEO_ENCODE_DPB_BIT_KHR |
+                                                VK_IMAGE_USAGE_SAMPLE_WEIGHT_BIT_QCOM | VK_IMAGE_USAGE_SAMPLE_BLOCK_MATCH_BIT_QCOM;
+    skip |= ValidateImageUsageFlags(VK_NULL_HANDLE, *image_state, valid_usage_flags, false,
+                                    "VUID-VkImageViewCreateInfo-image-04441", "vkCreateImageView()");
+    // If this isn't a sparse image, it needs to have memory backing it at CreateImageView time
+    skip |= ValidateMemoryIsBoundToImage(device, *image_state, "vkCreateImageView()", "VUID-VkImageViewCreateInfo-image-01020");
+    // Checks imported from image layer
+    skip |= ValidateCreateImageViewSubresourceRange(
+        image_state.get(), pCreateInfo->viewType == VK_IMAGE_VIEW_TYPE_2D || pCreateInfo->viewType == VK_IMAGE_VIEW_TYPE_2D_ARRAY,
```
Comment (ncesario-lunarg) [inline layers/core_checks/image_validation.cpp:1606]: Is this a clang-format-only change? If so, do we know how the file got in an "un-formatted state"?

File: layers/core_checks/image_validation.cpp [hunk 3]
```diff
@@ -1581,533 +1581,524 @@ bool CoreChecks::ValidateImageViewFormatFeatures(const IMAGE_STATE *image_state,
...
+    const VkFormat image_format = image_state->createInfo.format;
+    const VkFormat view_format = pCreateInfo->format;
+    const VkImageAspectFlags aspect_mask = pCreateInfo->subresourceRange.aspectMask;
+    const VkImageType image_type = image_state->createInfo.imageType;
+    const VkImageViewType view_type = pCreateInfo->viewType;
+    const uint32_t layer_count = pCreateInfo->subresourceRange.layerCount;
+
+    // If there's a chained VkImageViewUsageCreateInfo struct, modify image_usage to match
+    VkImageUsageFlags image_usage = image_state->createInfo.usage;
+    if (auto chained_ivuci_struct = LvlFindInChain<VkImageViewUsageCreateInfo>(pCreateInfo->pNext); chained_ivuci_struct) {
```
Comment (ncesario-lunarg) [inline layers/core_checks/image_validation.cpp:1620]: `const`?

**Review by ncesario-lunarg — APPROVED** [review]
Couple questions, but LGTM!

File: layers/core_checks/image_validation.cpp (hunk 2 above)
Comment (juan-lunarg) [reply layers/core_checks/image_validation.cpp:1606]: This was clang-format. I believe this is caused by the ColumnLimit of 132

File: layers/core_checks/image_validation.cpp (hunk 1 above)
Comment (juan-lunarg) [reply layers/core_checks/image_validation.cpp:1585]: Personal preference but I find [[maybe_unused]] easier to search for in a code base.

**Review by arno-lunarg — APPROVED** [review]
LGTM!

**Pull request merged**